cmake/gen_commit_h.cmake

```diff
@@ -26,6 +26,27 @@ if(GIT_FOUND)
 
   # Limit to 12 characters
   string(SUBSTRING "${${COMMIT_TYPE}_COMMIT}" 0 12 ${COMMIT_TYPE}_COMMIT)
+
+  execute_process(
+    COMMAND ${GIT_EXECUTABLE} describe --abbrev=1 --always --dirty
```

Comment:
this will not work as expected in all cases.
Please keep out of this PR and open separately for further discussions.

Reply:
in what cases will it not work?

Reply:
Has been removed